Analytics-API/MQTT.md

```diff
@@ -0,0 +1,116 @@
+---
+id: mqtt-integration
+title: MQTT Integration
+description: Guide on integrating PlaceOS with MQTT messaging
+---
+<!-- # MQTT -->
+
+<!-- images in old file, need rehosting https://docs.google.com/document/d/1gBZD296sF0cZXYyRKrRjn-wi5c408Yg1189S-DV9JgE/edit# -->
+
+PlaceOS supports publishing module state information via MQTT. 
+This provides environment information to external systems such as [Amazon MQTT Service](https://docs.aws.amazon.com/iot/latest/developerguide/view-mqtt-messages.html)
+
+MQTT messages consist of a *header* and a *payload* and typically have low bandwidth usage. 
+The header declares the topic of the message, and the payload carries data as key-value pairs.
+PlaceOS uses two types of message sent over MQTT: State Changes and Metadata.
+
+## State Change
+Changes to module state information propagates in real time. 
+All change messages share the following topic structure:
+
+``` html
+placeos/<org>/state/<bld>/<lvl>/<area>/<sys>/<drv>/<mod>/<idx>/<state>
+```
+In this structure, each section is a unique identifying tag to represent part of the system.
+- `<org>`: Organization ID
+- `<bld>`: Building ID
+- `<lvl>`: Level ID
+- `<area>`: [Area](../key-concepts/zones.md) ID
+- `<sys>`: [System](../key-concepts/systems.md) ID
+- `<drv>`: [Driver](../key-concepts/drivers.md) ID
+- `<mod>`: [Module](../key-concepts/modules.md) ID
+- `<idx>`: Module Index
+- `<state>`: State Key
+
+<!-- some kind of key expanding the above. Original had an image. html isn't *right* but the colouratin is nice-->
+
+<!--What part of PlaceOS? messaging service? Does this microservice (?) have a title? -->
+On a state change, PlaceOS  will publish a message with the payload containing the new state value as a JSON entity. 
+The associated driver defines the structure and change frequency of this state.
+If a state change is for a system which isn't assigned to a building, level or area, that topic level will be an underscore character 
+ (`“_”`).
```

Comment:
```suggestion
 (`_`).
```

Anything that is a literal can just be specified directly as the char / statement in backticks.

Super minor nitpick, but this should also likely be part of the above line.

Reply:
Yep, using ticks and quotes was an error. And I think one of the rules (complexity) was cranky about it being a line up before, but as it's a warning now it can be noted and moved on